### ISSUE TITLE
local reference table overflow

### ISSUE THREAD

**Issue by joshatcaper** [emanuele-f/PCAPdroid#7]
I've been trying to use this APK but I've been running into a strange issue. I'm getting this stack trace:

```
2020-...art/runtime/indirect_reference_table.cc:132] JNI ERROR (app bug): local reference table overflow (max=512)
2020-...art/runtime/indirect_reference_table.cc:132] local reference table dump:
2020-...art/runtime/indirect_reference_table.cc:132]   Last 10 entries (of 512):
2020-...art/runtime/indirect_reference_table.cc:132]       511: 0x12ded740 java.lang.String ""
2020-...art/runtime/indirect_reference_table.cc:132]       510: 0x12df90e8 com.emanuelef.remote_capture.ConnDescriptor[] (3 elements)
2020-...art/runtime/indirect_reference_table.cc:132]       509: 0x12ca9d30 java.lang.Class<com.emanuelef.remote_capture.ConnDescriptor>
2020-...art/runtime/indirect_reference_table.cc:132]       508: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/indirect_reference_table.cc:132]       507: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/indirect_reference_table.cc:132]       506: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/indirect_reference_table.cc:132]       505: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/indirect_reference_table.cc:132]       504: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/indirect_reference_table.cc:132]       503: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/indirect_reference_table.cc:132]       502: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/indirect_reference_table.cc:132]   Summary:
2020-...art/runtime/indirect_reference_table.cc:132]        53 of com.emanuelef.remote_capture.ConnDescriptor (53 unique instances)
2020-...art/runtime/indirect_reference_table.cc:132]         1 of com.emanuelef.remote_capture.ConnDescriptor[]
2020-...art/runtime/indirect_reference_table.cc:132]         5 of com.emanuelef.remote_capture.ConnDescriptor[] (2 elements) (5 unique instances)
2020-...art/runtime/indirect_reference_table.cc:132]        14 of com.emanuelef.remote_capture.ConnDescriptor[] (3 elements) (14 unique instances)
2020-...art/runtime/indirect_reference_table.cc:132]         1 of com.emanuelef.remote_capture.ConnDescriptor[] (4 elements)
2020-...art/runtime/indirect_reference_table.cc:132]         1 of java.lang.Thread
2020-...art/runtime/indirect_reference_table.cc:132]       274 of java.lang.String (274 unique instances)
2020-...art/runtime/indirect_reference_table.cc:132]       163 of java.lang.Class (2 unique instances)
2020-...art/runtime/indirect_reference_table.cc:132] 
2020-...art/runtime/runtime.cc:422] Runtime aborting...
2020-...art/runtime/runtime.cc:422] Aborting thread:
2020-...art/runtime/runtime.cc:422] "CaptureService Thread" prio=5 tid=17 Runnable
2020-...art/runtime/runtime.cc:422]   | group="" sCount=0 dsCount=0 obj=0x12d25160 self=0x7f18b21400
2020-...art/runtime/runtime.cc:422]   | sysTid=4480 nice=0 cgrp=default sched=0/0 handle=0x7f153ae450
2020-...art/runtime/runtime.cc:422]   | state=R schedstat=( 6287098280 1315732660 5421 ) utm=43 stm=584 core=4 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f152ac000-0x7f152ae000 stackSize=1037KB
2020-...art/runtime/runtime.cc:422]   | held mutexes= "abort lock" "mutator lock"(shared held)
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000047e090  /system/lib64/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiP12BacktraceMapPKcPNS_9ArtMethodEPv+220)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 000000000047e08c  /system/lib64/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiP12BacktraceMapPKcPNS_9ArtMethodEPv+216)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 00000000004521f8  /system/lib64/libart.so (_ZNK3art6Thread9DumpStackERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEEbP12BacktraceMap+480)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 0000000000440170  /system/lib64/libart.so (_ZNK3art10AbortState10DumpThreadERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEEPNS_6ThreadE+56)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 000000000043ff90  /system/lib64/libart.so (_ZNK3art10AbortState4DumpERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEE+576)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 0000000000433998  /system/lib64/libart.so (_ZN3art7Runtime5AbortEPKc+148)
2020-...art/runtime/runtime.cc:422]   native: #06 pc 00000000000e5a7c  /system/lib64/libart.so (_ZN3art10LogMessageD2Ev+1592)
2020-...art/runtime/runtime.cc:422]   native: #07 pc 000000000024d918  /system/lib64/libart.so (_ZN3art22IndirectReferenceTable3AddEjPNS_6mirror6ObjectE+308)
2020-...art/runtime/runtime.cc:422]   native: #08 pc 000000000034c78c  /system/lib64/libart.so (_ZN3art3JNI12NewStringUTFEP7_JNIEnvPKc+616)
2020-...art/runtime/runtime.cc:422]   native: #09 pc 0000000000003a90  /data/app/com.emanuelef.remote_capture-1/lib/arm64/libvpnproxy-jni.so (???)
2020-...art/runtime/runtime.cc:422]   native: #10 pc 00000000000034b0  /data/app/com.emanuelef.remote_capture-1/lib/arm64/libzdtun.so (zdtun_iter_connections+108)
2020-...art/runtime/runtime.cc:422]   at com.emanuelef.remote_capture.CaptureService.runPacketLoop(Native method)
2020-...art/runtime/runtime.cc:422]   at com.emanuelef.remote_capture.CaptureService.run(CaptureService.java:235)
2020-...art/runtime/runtime.cc:422]   at java.lang.Thread.run(Thread.java:761)
2020-...art/runtime/runtime.cc:422] Dumping all threads without appropriate locks held: thread list lock
2020-...art/runtime/runtime.cc:422] All threads:
2020-...art/runtime/runtime.cc:422] DALVIK THREADS (17):
2020-...art/runtime/runtime.cc:422] "CaptureService Thread" prio=5 tid=17 Runnable
2020-...art/runtime/runtime.cc:422]   | group="" sCount=0 dsCount=0 obj=0x12d25160 self=0x7f18b21400
2020-...art/runtime/runtime.cc:422]   | sysTid=4480 nice=0 cgrp=default sched=0/0 handle=0x7f153ae450
2020-...art/runtime/runtime.cc:422]   | state=R schedstat=( 6307026989 1315763576 5423 ) utm=44 stm=585 core=4 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f152ac000-0x7f152ae000 stackSize=1037KB
2020-...art/runtime/runtime.cc:422]   | held mutexes= "abort lock" "mutator lock"(shared held)
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000047e090  /system/lib64/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiP12BacktraceMapPKcPNS_9ArtMethodEPv+220)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 000000000047e08c  /system/lib64/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiP12BacktraceMapPKcPNS_9ArtMethodEPv+216)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 00000000004521f8  /system/lib64/libart.so (_ZNK3art6Thread9DumpStackERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEEbP12BacktraceMap+480)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 0000000000469d7c  /system/lib64/libart.so (_ZN3art14DumpCheckpoint3RunEPNS_6ThreadE+832)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 0000000000461f64  /system/lib64/libart.so (_ZN3art10ThreadList13RunCheckpointEPNS_7ClosureE+476)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 0000000000461b54  /system/lib64/libart.so (_ZN3art10ThreadList4DumpERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEEb+848)
2020-...art/runtime/runtime.cc:422]   native: #06 pc 000000000043ffa0  /system/lib64/libart.so (_ZNK3art10AbortState4DumpERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEE+592)
2020-...art/runtime/runtime.cc:422]   native: #07 pc 0000000000433998  /system/lib64/libart.so (_ZN3art7Runtime5AbortEPKc+148)
2020-...art/runtime/runtime.cc:422]   native: #08 pc 00000000000e5a7c  /system/lib64/libart.so (_ZN3art10LogMessageD2Ev+1592)
2020-...art/runtime/runtime.cc:422]   native: #09 pc 000000000024d918  /system/lib64/libart.so (_ZN3art22IndirectReferenceTable3AddEjPNS_6mirror6ObjectE+308)
2020-...art/runtime/runtime.cc:422]   native: #10 pc 000000000034c78c  /system/lib64/libart.so (_ZN3art3JNI12NewStringUTFEP7_JNIEnvPKc+616)
2020-...art/runtime/runtime.cc:422]   native: #11 pc 0000000000003a90  /data/app/com.emanuelef.remote_capture-1/lib/arm64/libvpnproxy-jni.so (???)
2020-...art/runtime/runtime.cc:422]   native: #12 pc 00000000000034b0  /data/app/com.emanuelef.remote_capture-1/lib/arm64/libzdtun.so (zdtun_iter_connections+108)
2020-...art/runtime/runtime.cc:422]   at com.emanuelef.remote_capture.CaptureService.runPacketLoop(Native method)
2020-...art/runtime/runtime.cc:422]   at com.emanuelef.remote_capture.CaptureService.run(CaptureService.java:235)
2020-...art/runtime/runtime.cc:422]   at java.lang.Thread.run(Thread.java:761)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "main" prio=5 tid=1 Native
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x753d2850 self=0x7f33895a00
2020-...art/runtime/runtime.cc:422]   | sysTid=4434 nice=0 cgrp=default sched=0/0 handle=0x7f37911a98
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 652039840 66308616 670 ) utm=53 stm=11 core=3 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7fc61dc000-0x7fc61de000 stackSize=8MB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: SyS_epoll_wait+0x2cc/0x384
2020-...art/runtime/runtime.cc:422]   kernel: SyS_epoll_pwait+0xb4/0x12c
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000006ab80  /system/lib64/libc.so (__epoll_pwait+8)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 000000000001e1a4  /system/lib64/libc.so (epoll_pwait+64)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 0000000000018010  /system/lib64/libutils.so (_ZN7android6Looper9pollInnerEi+156)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 0000000000017eb4  /system/lib64/libutils.so (_ZN7android6Looper8pollOnceEiPiS1_PPv+60)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 00000000000f05f0  /system/lib64/libandroid_runtime.so (_ZN7android18NativeMessageQueue8pollOnceEP7_JNIEnvP8_jobjecti+48)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 00000000008a3e40  /system/framework/arm64/boot-framework.oat (Java_android_os_MessageQueue_nativePollOnce__JI+140)
2020-...art/runtime/runtime.cc:422]   at android.os.MessageQueue.nativePollOnce(Native method)
2020-...art/runtime/runtime.cc:422]   at android.os.MessageQueue.next(MessageQueue.java:323)
2020-...art/runtime/runtime.cc:422]   at android.os.Looper.loop(Looper.java:136)
2020-...art/runtime/runtime.cc:422]   at android.app.ActivityThread.main(ActivityThread.java:6119)
2020-...art/runtime/runtime.cc:422]   at java.lang.reflect.Method.invoke!(Native method)
2020-...art/runtime/runtime.cc:422]   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:902)
2020-...art/runtime/runtime.cc:422]   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:792)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "Jit thread pool worker thread 0" prio=5 tid=2 Native (still starting up)
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x0 self=0x7f2c00d000
2020-...art/runtime/runtime.cc:422]   | sysTid=4439 nice=9 cgrp=default sched=0/0 handle=0x7f32f04450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 153258252 10977789 45 ) utm=12 stm=2 core=2 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f32e06000-0x7f32e08000 stackSize=1021KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait_queue_me+0xd8/0x144
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait+0xec/0x200
2020-...art/runtime/runtime.cc:422]   kernel: do_futex+0xdc/0x49c
2020-...art/runtime/runtime.cc:422]   kernel: SyS_futex+0x110/0x19c
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000001bcec  /system/lib64/libc.so (syscall+28)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 00000000000e7dcc  /system/lib64/libart.so (_ZN3art17ConditionVariable16WaitHoldingLocksEPNS_6ThreadE+156)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 000000000046bb24  /system/lib64/libart.so (_ZN3art10ThreadPool7GetTaskEPNS_6ThreadE+248)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 000000000046afcc  /system/lib64/libart.so (_ZN3art16ThreadPoolWorker3RunEv+124)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 000000000046a8f0  /system/lib64/libart.so (_ZN3art16ThreadPoolWorker8CallbackEPv+116)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 0000000000068748  /system/lib64/libc.so (_ZL15__pthread_startPv+208)
2020-...art/runtime/runtime.cc:422]   native: #06 pc 000000000001da7c  /system/lib64/libc.so (__start_thread+16)
2020-...art/runtime/runtime.cc:422]   (no managed stack frames)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "Signal Catcher" prio=5 tid=3 WaitingInMainSignalCatcherLoop
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12c44670 self=0x7f33896400
2020-...art/runtime/runtime.cc:422]   | sysTid=4440 nice=0 cgrp=default sched=0/0 handle=0x7f32e03450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 847250 0 1 ) utm=0 stm=0 core=2 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f32d09000-0x7f32d0b000 stackSize=1005KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: do_sigtimedwait+0xe8/0x1cc
2020-...art/runtime/runtime.cc:422]   kernel: SyS_rt_sigtimedwait+0xc8/0x118
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000006ad60  /system/lib64/libc.so (__rt_sigtimedwait+8)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 0000000000024d48  /system/lib64/libc.so (sigwait+64)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 00000000004448d4  /system/lib64/libart.so (_ZN3art9SignalSet4WaitEv+48)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 0000000000444390  /system/lib64/libart.so (_ZN3art13SignalCatcher13WaitForSignalEPNS_6ThreadERNS_9SignalSetE+232)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 0000000000442924  /system/lib64/libart.so (_ZN3art13SignalCatcher3RunEPv+400)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 0000000000068748  /system/lib64/libc.so (_ZL15__pthread_startPv+208)
2020-...art/runtime/runtime.cc:422]   native: #06 pc 000000000001da7c  /system/lib64/libc.so (__start_thread+16)
2020-...art/runtime/runtime.cc:422]   (no managed stack frames)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "JDWP" prio=5 tid=4 WaitingInMainDebuggerLoop
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12c44700 self=0x7f2c016400
2020-...art/runtime/runtime.cc:422]   | sysTid=4441 nice=0 cgrp=default sched=0/0 handle=0x7f32d06450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 2313751 860708 11 ) utm=0 stm=0 core=2 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f32c0c000-0x7f32c0e000 stackSize=1005KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: poll_schedule_timeout+0x50/0x84
2020-...art/runtime/runtime.cc:422]   kernel: do_select+0x4b8/0x510
2020-...art/runtime/runtime.cc:422]   kernel: core_sys_select+0x234/0x390
2020-...art/runtime/runtime.cc:422]   kernel: SyS_pselect6+0x180/0x218
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000006acb8  /system/lib64/libc.so (__pselect6+8)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 0000000000023244  /system/lib64/libc.so (select+156)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 0000000000557324  /system/lib64/libart.so (_ZN3art4JDWP12JdwpAdbState15ProcessIncomingEv+348)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 00000000003056c0  /system/lib64/libart.so (_ZN3art4JDWP9JdwpState3RunEv+916)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 0000000000304b70  /system/lib64/libart.so (_ZN3art4JDWPL15StartJdwpThreadEPv+48)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 0000000000068748  /system/lib64/libc.so (_ZL15__pthread_startPv+208)
2020-...art/runtime/runtime.cc:422]   native: #06 pc 000000000001da7c  /system/lib64/libc.so (__start_thread+16)
2020-...art/runtime/runtime.cc:422]   (no managed stack frames)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "ReferenceQueueDaemon" prio=5 tid=5 Waiting
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12c44790 self=0x7f3386a000
2020-...art/runtime/runtime.cc:422]   | sysTid=4442 nice=0 cgrp=default sched=0/0 handle=0x7f32c09450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 921918 120168 12 ) utm=0 stm=0 core=2 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f32b07000-0x7f32b09000 stackSize=1037KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait_queue_me+0xd8/0x144
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait+0xec/0x200
2020-...art/runtime/runtime.cc:422]   kernel: do_futex+0xdc/0x49c
2020-...art/runtime/runtime.cc:422]   kernel: SyS_futex+0x110/0x19c
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000001bcec  /system/lib64/libc.so (syscall+28)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 00000000000e7dcc  /system/lib64/libart.so (_ZN3art17ConditionVariable16WaitHoldingLocksEPNS_6ThreadE+156)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 000000000037c49c  /system/lib64/libart.so (_ZN3art7Monitor4WaitEPNS_6ThreadElibNS_11ThreadStateE+660)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 0000000000000810  /system/framework/arm64/boot.oat (Java_java_lang_Object_wait__+124)
2020-...art/runtime/runtime.cc:422]   at java.lang.Object.wait!(Native method)
2020-...art/runtime/runtime.cc:422]   - waiting on <0x0762088f> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
2020-...art/runtime/runtime.cc:422]   at java.lang.Daemons$ReferenceQueueDaemon.run(Daemons.java:150)
2020-...art/runtime/runtime.cc:422]   - locked <0x0762088f> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
2020-...art/runtime/runtime.cc:422]   at java.lang.Thread.run(Thread.java:761)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "FinalizerDaemon" prio=5 tid=6 Waiting
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12c44820 self=0x7f3386aa00
2020-...art/runtime/runtime.cc:422]   | sysTid=4443 nice=0 cgrp=default sched=0/0 handle=0x7f32b04450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 928957 7974417 7 ) utm=0 stm=0 core=2 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f32a02000-0x7f32a04000 stackSize=1037KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait_queue_me+0xd8/0x144
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait+0xec/0x200
2020-...art/runtime/runtime.cc:422]   kernel: do_futex+0xdc/0x49c
2020-...art/runtime/runtime.cc:422]   kernel: SyS_futex+0x110/0x19c
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000001bcec  /system/lib64/libc.so (syscall+28)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 00000000000e7dcc  /system/lib64/libart.so (_ZN3art17ConditionVariable16WaitHoldingLocksEPNS_6ThreadE+156)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 000000000037c49c  /system/lib64/libart.so (_ZN3art7Monitor4WaitEPNS_6ThreadElibNS_11ThreadStateE+660)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 0000000000000980  /system/framework/arm64/boot.oat (Java_java_lang_Object_wait__JI+140)
2020-...art/runtime/runtime.cc:422]   at java.lang.Object.wait!(Native method)
2020-...art/runtime/runtime.cc:422]   - waiting on <0x00f3671c> (a java.lang.Object)
2020-...art/runtime/runtime.cc:422]   at java.lang.Object.wait(Object.java:407)
2020-...art/runtime/runtime.cc:422]   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:188)
2020-...art/runtime/runtime.cc:422]   - locked <0x00f3671c> (a java.lang.Object)
2020-...art/runtime/runtime.cc:422]   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:209)
2020-...art/runtime/runtime.cc:422]   at java.lang.Daemons$FinalizerDaemon.run(Daemons.java:204)
2020-...art/runtime/runtime.cc:422]   at java.lang.Thread.run(Thread.java:761)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "FinalizerWatchdogDaemon" prio=5 tid=7 Waiting
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12c448b0 self=0x7f3386b400
2020-...art/runtime/runtime.cc:422]   | sysTid=4444 nice=0 cgrp=default sched=0/0 handle=0x7f329ff450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 178458 409792 2 ) utm=0 stm=0 core=2 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f328fd000-0x7f328ff000 stackSize=1037KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait_queue_me+0xd8/0x144
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait+0xec/0x200
2020-...art/runtime/runtime.cc:422]   kernel: do_futex+0xdc/0x49c
2020-...art/runtime/runtime.cc:422]   kernel: SyS_futex+0x110/0x19c
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000001bcec  /system/lib64/libc.so (syscall+28)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 00000000000e7dcc  /system/lib64/libart.so (_ZN3art17ConditionVariable16WaitHoldingLocksEPNS_6ThreadE+156)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 000000000037c49c  /system/lib64/libart.so (_ZN3art7Monitor4WaitEPNS_6ThreadElibNS_11ThreadStateE+660)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 0000000000000810  /system/framework/arm64/boot.oat (Java_java_lang_Object_wait__+124)
2020-...art/runtime/runtime.cc:422]   at java.lang.Object.wait!(Native method)
2020-...art/runtime/runtime.cc:422]   - waiting on <0x0cb59f25> (a java.lang.Daemons$FinalizerWatchdogDaemon)
2020-...art/runtime/runtime.cc:422]   at java.lang.Daemons$FinalizerWatchdogDaemon.sleepUntilNeeded(Daemons.java:269)
2020-...art/runtime/runtime.cc:422]   - locked <0x0cb59f25> (a java.lang.Daemons$FinalizerWatchdogDaemon)
2020-...art/runtime/runtime.cc:422]   at java.lang.Daemons$FinalizerWatchdogDaemon.run(Daemons.java:249)
2020-...art/runtime/runtime.cc:422]   at java.lang.Thread.run(Thread.java:761)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "Binder:4434_1" prio=5 tid=8 Native
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12c44af0 self=0x7f2c027800
2020-...art/runtime/runtime.cc:422]   | sysTid=4446 nice=0 cgrp=default sched=0/0 handle=0x7f326f7450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 5998415 7011460 18 ) utm=0 stm=0 core=5 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f325fd000-0x7f325ff000 stackSize=1005KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: binder_thread_read+0xdd8/0xfb8
2020-...art/runtime/runtime.cc:422]   kernel: binder_ioctl_write_read+0x1ac/0x2f8
2020-...art/runtime/runtime.cc:422]   kernel: binder_ioctl+0x278/0x6ac
2020-...art/runtime/runtime.cc:422]   kernel: do_vfs_ioctl+0x4d0/0x5c0
2020-...art/runtime/runtime.cc:422]   kernel: SyS_ioctl+0x60/0x88
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000006ac6c  /system/lib64/libc.so (__ioctl+4)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 000000000001fab4  /system/lib64/libc.so (ioctl+140)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 0000000000055568  /system/lib64/libbinder.so (_ZN7android14IPCThreadState14talkWithDriverEb+264)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 00000000000556cc  /system/lib64/libbinder.so (_ZN7android14IPCThreadState20getAndExecuteCommandEv+24)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 0000000000055e04  /system/lib64/libbinder.so (_ZN7android14IPCThreadState14joinThreadPoolEb+72)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 0000000000072d60  /system/lib64/libbinder.so (???)
2020-...art/runtime/runtime.cc:422]   native: #06 pc 000000000001245c  /system/lib64/libutils.so (_ZN7android6Thread11_threadLoopEPv+272)
2020-...art/runtime/runtime.cc:422]   native: #07 pc 000000000009eeb0  /system/lib64/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+116)
2020-...art/runtime/runtime.cc:422]   native: #08 pc 0000000000068748  /system/lib64/libc.so (_ZL15__pthread_startPv+208)
2020-...art/runtime/runtime.cc:422]   native: #09 pc 000000000001da7c  /system/lib64/libc.so (__start_thread+16)
2020-...art/runtime/runtime.cc:422]   (no managed stack frames)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "HeapTaskDaemon" prio=5 tid=9 Blocked
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12c44940 self=0x7f3386be00
2020-...art/runtime/runtime.cc:422]   | sysTid=4445 nice=0 cgrp=default sched=0/0 handle=0x7f328fa450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 14572538 10085543 26 ) utm=1 stm=0 core=4 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f327f8000-0x7f327fa000 stackSize=1037KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait_queue_me+0xd8/0x144
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait+0xec/0x200
2020-...art/runtime/runtime.cc:422]   kernel: do_futex+0xdc/0x49c
2020-...art/runtime/runtime.cc:422]   kernel: SyS_futex+0x110/0x19c
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000001bcec  /system/lib64/libc.so (syscall+28)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 00000000000e7dcc  /system/lib64/libart.so (_ZN3art17ConditionVariable16WaitHoldingLocksEPNS_6ThreadE+156)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 00000000002427f8  /system/lib64/libart.so (_ZN3art2gc13TaskProcessor7GetTaskEPNS_6ThreadE+276)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 0000000000243130  /system/lib64/libart.so (_ZN3art2gc13TaskProcessor11RunAllTasksEPNS_6ThreadE+92)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 00000000001e6ac0  /system/framework/arm64/boot-core-libart.oat (Java_dalvik_system_VMRuntime_runHeapTasks__+124)
2020-...art/runtime/runtime.cc:422]   at dalvik.system.VMRuntime.runHeapTasks(Native method)
2020-...art/runtime/runtime.cc:422]   - waiting to lock an unknown object
2020-...art/runtime/runtime.cc:422]   at java.lang.Daemons$HeapTaskDaemon.run(Daemons.java:433)
2020-...art/runtime/runtime.cc:422]   at java.lang.Thread.run(Thread.java:761)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "Binder:4434_2" prio=5 tid=10 Native
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12c44b80 self=0x7f2c02dc00
2020-...art/runtime/runtime.cc:422]   | sysTid=4447 nice=0 cgrp=default sched=0/0 handle=0x7f325fa450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 5146749 1634500 25 ) utm=0 stm=0 core=1 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f32500000-0x7f32502000 stackSize=1005KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: binder_thread_read+0xdd8/0xfb8
2020-...art/runtime/runtime.cc:422]   kernel: binder_ioctl_write_read+0x1ac/0x2f8
2020-...art/runtime/runtime.cc:422]   kernel: binder_ioctl+0x278/0x6ac
2020-...art/runtime/runtime.cc:422]   kernel: do_vfs_ioctl+0x4d0/0x5c0
2020-...art/runtime/runtime.cc:422]   kernel: SyS_ioctl+0x60/0x88
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000006ac6c  /system/lib64/libc.so (__ioctl+4)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 000000000001fab4  /system/lib64/libc.so (ioctl+140)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 0000000000055568  /system/lib64/libbinder.so (_ZN7android14IPCThreadState14talkWithDriverEb+264)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 00000000000556cc  /system/lib64/libbinder.so (_ZN7android14IPCThreadState20getAndExecuteCommandEv+24)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 0000000000055e04  /system/lib64/libbinder.so (_ZN7android14IPCThreadState14joinThreadPoolEb+72)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 0000000000072d60  /system/lib64/libbinder.so (???)
2020-...art/runtime/runtime.cc:422]   native: #06 pc 000000000001245c  /system/lib64/libutils.so (_ZN7android6Thread11_threadLoopEPv+272)
2020-...art/runtime/runtime.cc:422]   native: #07 pc 000000000009eeb0  /system/lib64/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+116)
2020-...art/runtime/runtime.cc:422]   native: #08 pc 0000000000068748  /system/lib64/libc.so (_ZL15__pthread_startPv+208)
2020-...art/runtime/runtime.cc:422]   native: #09 pc 000000000001da7c  /system/lib64/libc.so (__start_thread+16)
2020-...art/runtime/runtime.cc:422]   (no managed stack frames)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "Profile Saver" prio=5 tid=11 Native
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12c44f70 self=0x7f3386c800
2020-...art/runtime/runtime.cc:422]   | sysTid=4448 nice=-10 cgrp=default sched=0/0 handle=0x7f324f5450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 13715042 292291 11 ) utm=1 stm=0 core=1 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f323fb000-0x7f323fd000 stackSize=1005KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait_queue_me+0xd8/0x144
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait+0xec/0x200
2020-...art/runtime/runtime.cc:422]   kernel: do_futex+0xdc/0x49c
2020-...art/runtime/runtime.cc:422]   kernel: SyS_futex+0x110/0x19c
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000001bcec  /system/lib64/libc.so (syscall+28)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 00000000000e7dcc  /system/lib64/libart.so (_ZN3art17ConditionVariable16WaitHoldingLocksEPNS_6ThreadE+156)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 000000000031ea04  /system/lib64/libart.so (_ZN3art12ProfileSaver3RunEv+284)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 000000000032013c  /system/lib64/libart.so (_ZN3art12ProfileSaver21RunProfileSaverThreadEPv+100)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 0000000000068748  /system/lib64/libc.so (_ZL15__pthread_startPv+208)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 000000000001da7c  /system/lib64/libc.so (__start_thread+16)
2020-...art/runtime/runtime.cc:422]   (no managed stack frames)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "ModernAsyncTask #1" prio=5 tid=12 Waiting
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12d52b80 self=0x7f295bcc00
2020-...art/runtime/runtime.cc:422]   | sysTid=4452 nice=10 cgrp=bg_non_interactive sched=0/0 handle=0x7f2c4a2450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 34034000 12822583 59 ) utm=1 stm=1 core=4 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f2c3a0000-0x7f2c3a2000 stackSize=1037KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait_queue_me+0xd8/0x144
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait+0xec/0x200
2020-...art/runtime/runtime.cc:422]   kernel: do_futex+0xdc/0x49c
2020-...art/runtime/runtime.cc:422]   kernel: SyS_futex+0x110/0x19c
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000001bcec  /system/lib64/libc.so (syscall+28)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 00000000000e7dcc  /system/lib64/libart.so (_ZN3art17ConditionVariable16WaitHoldingLocksEPNS_6ThreadE+156)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 000000000037c49c  /system/lib64/libart.so (_ZN3art7Monitor4WaitEPNS_6ThreadElibNS_11ThreadStateE+660)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 0000000000000980  /system/framework/arm64/boot.oat (Java_java_lang_Object_wait__JI+140)
2020-...art/runtime/runtime.cc:422]   at java.lang.Object.wait!(Native method)
2020-...art/runtime/runtime.cc:422]   - waiting on <0x0b7038fa> (a java.lang.Object)
2020-...art/runtime/runtime.cc:422]   at java.lang.Thread.parkFor$(Thread.java:2127)
2020-...art/runtime/runtime.cc:422]   - locked <0x0b7038fa> (a java.lang.Object)
2020-...art/runtime/runtime.cc:422]   at sun.misc.Unsafe.park(Unsafe.java:325)
2020-...art/runtime/runtime.cc:422]   at java.util.concurrent.locks.LockSupport.park(LockSupport.java:161)
2020-...art/runtime/runtime.cc:422]   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2035)
2020-...art/runtime/runtime.cc:422]   at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:413)
2020-...art/runtime/runtime.cc:422]   at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1058)
2020-...art/runtime/runtime.cc:422]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1118)
2020-...art/runtime/runtime.cc:422]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
2020-...art/runtime/runtime.cc:422]   at java.lang.Thread.run(Thread.java:761)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "RenderThread" prio=5 tid=13 Native
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12cf9dc0 self=0x7f18ab3800
2020-...art/runtime/runtime.cc:422]   | sysTid=4453 nice=0 cgrp=default sched=0/0 handle=0x7f187ff450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 355637056 31252585 787 ) utm=24 stm=10 core=5 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f18705000-0x7f18707000 stackSize=1005KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: SyS_epoll_wait+0x2cc/0x384
2020-...art/runtime/runtime.cc:422]   kernel: SyS_epoll_pwait+0xb4/0x12c
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000006ab80  /system/lib64/libc.so (__epoll_pwait+8)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 000000000001e1a4  /system/lib64/libc.so (epoll_pwait+64)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 0000000000018010  /system/lib64/libutils.so (_ZN7android6Looper9pollInnerEi+156)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 0000000000017eb4  /system/lib64/libutils.so (_ZN7android6Looper8pollOnceEiPiS1_PPv+60)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 000000000003ac98  /system/lib64/libhwui.so (_ZN7android10uirenderer12renderthread12RenderThread10threadLoopEv+664)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 000000000001245c  /system/lib64/libutils.so (_ZN7android6Thread11_threadLoopEPv+272)
2020-...art/runtime/runtime.cc:422]   native: #06 pc 000000000009eeb0  /system/lib64/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+116)
2020-...art/runtime/runtime.cc:422]   native: #07 pc 0000000000068748  /system/lib64/libc.so (_ZL15__pthread_startPv+208)
2020-...art/runtime/runtime.cc:422]   native: #08 pc 000000000001da7c  /system/lib64/libc.so (__start_thread+16)
2020-...art/runtime/runtime.cc:422]   (no managed stack frames)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "hwuiTask2" prio=5 tid=14 Native
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12d255e0 self=0x7f189df000
2020-...art/runtime/runtime.cc:422]   | sysTid=4465 nice=-2 cgrp=default sched=0/0 handle=0x7f1723c450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 6521626 2698416 52 ) utm=0 stm=0 core=1 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f17142000-0x7f17144000 stackSize=1005KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait_queue_me+0xd8/0x144
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait+0xec/0x200
2020-...art/runtime/runtime.cc:422]   kernel: do_futex+0xdc/0x49c
2020-...art/runtime/runtime.cc:422]   kernel: SyS_futex+0x110/0x19c
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000001bcec  /system/lib64/libc.so (syscall+28)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 0000000000067e78  /system/lib64/libc.so (pthread_cond_wait+96)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 000000000003bb54  /system/lib64/libhwui.so (???)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 000000000001245c  /system/lib64/libutils.so (_ZN7android6Thread11_threadLoopEPv+272)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 000000000009eeb0  /system/lib64/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+116)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 0000000000068748  /system/lib64/libc.so (_ZL15__pthread_startPv+208)
2020-...art/runtime/runtime.cc:422]   native: #06 pc 000000000001da7c  /system/lib64/libc.so (__start_thread+16)
2020-...art/runtime/runtime.cc:422]   (no managed stack frames)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "hwuiTask1" prio=5 tid=15 Native
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12d25550 self=0x7f1799b000
2020-...art/runtime/runtime.cc:422]   | sysTid=4464 nice=-2 cgrp=default sched=0/0 handle=0x7f17339450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 12443959 8062332 91 ) utm=0 stm=0 core=1 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f1723f000-0x7f17241000 stackSize=1005KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait_queue_me+0xd8/0x144
2020-...art/runtime/runtime.cc:422]   kernel: futex_wait+0xec/0x200
2020-...art/runtime/runtime.cc:422]   kernel: do_futex+0xdc/0x49c
2020-...art/runtime/runtime.cc:422]   kernel: SyS_futex+0x110/0x19c
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000001bcec  /system/lib64/libc.so (syscall+28)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 0000000000067e78  /system/lib64/libc.so (pthread_cond_wait+96)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 000000000003bb54  /system/lib64/libhwui.so (???)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 000000000001245c  /system/lib64/libutils.so (_ZN7android6Thread11_threadLoopEPv+272)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 000000000009eeb0  /system/lib64/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+116)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 0000000000068748  /system/lib64/libc.so (_ZL15__pthread_startPv+208)
2020-...art/runtime/runtime.cc:422]   native: #06 pc 000000000001da7c  /system/lib64/libc.so (__start_thread+16)
2020-...art/runtime/runtime.cc:422]   (no managed stack frames)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] "Binder:4434_3" prio=5 tid=16 Native
2020-...art/runtime/runtime.cc:422]   | group="" sCount=1 dsCount=0 obj=0x12d254c0 self=0x7f189e5400
2020-...art/runtime/runtime.cc:422]   | sysTid=4466 nice=0 cgrp=default sched=0/0 handle=0x7f15949450
2020-...art/runtime/runtime.cc:422]   | state=S schedstat=( 2590874 2143168 9 ) utm=0 stm=0 core=1 HZ=100
2020-...art/runtime/runtime.cc:422]   | stack=0x7f1584f000-0x7f15851000 stackSize=1005KB
2020-...art/runtime/runtime.cc:422]   | held mutexes=
2020-...art/runtime/runtime.cc:422]   kernel: __switch_to+0xb4/0xc0
2020-...art/runtime/runtime.cc:422]   kernel: binder_thread_read+0xdd8/0xfb8
2020-...art/runtime/runtime.cc:422]   kernel: binder_ioctl_write_read+0x1ac/0x2f8
2020-...art/runtime/runtime.cc:422]   kernel: binder_ioctl+0x278/0x6ac
2020-...art/runtime/runtime.cc:422]   kernel: do_vfs_ioctl+0x4d0/0x5c0
2020-...art/runtime/runtime.cc:422]   kernel: SyS_ioctl+0x60/0x88
2020-...art/runtime/runtime.cc:422]   kernel: el0_svc_naked+0x24/0x28
2020-...art/runtime/runtime.cc:422]   native: #00 pc 000000000006ac6c  /system/lib64/libc.so (__ioctl+4)
2020-...art/runtime/runtime.cc:422]   native: #01 pc 000000000001fab4  /system/lib64/libc.so (ioctl+140)
2020-...art/runtime/runtime.cc:422]   native: #02 pc 0000000000055568  /system/lib64/libbinder.so (_ZN7android14IPCThreadState14talkWithDriverEb+264)
2020-...art/runtime/runtime.cc:422]   native: #03 pc 00000000000556cc  /system/lib64/libbinder.so (_ZN7android14IPCThreadState20getAndExecuteCommandEv+24)
2020-...art/runtime/runtime.cc:422]   native: #04 pc 0000000000055e04  /system/lib64/libbinder.so (_ZN7android14IPCThreadState14joinThreadPoolEb+72)
2020-...art/runtime/runtime.cc:422]   native: #05 pc 0000000000072d60  /system/lib64/libbinder.so (???)
2020-...art/runtime/runtime.cc:422]   native: #06 pc 000000000001245c  /system/lib64/libutils.so (_ZN7android6Thread11_threadLoopEPv+272)
2020-...art/runtime/runtime.cc:422]   native: #07 pc 000000000009eeb0  /system/lib64/libandroid_runtime.so (_ZN7android14AndroidRuntime15javaThreadShellEPv+116)
2020-...art/runtime/runtime.cc:422]   native: #08 pc 0000000000068748  /system/lib64/libc.so (_ZL15__pthread_startPv+208)
2020-...art/runtime/runtime.cc:422]   native: #09 pc 000000000001da7c  /system/lib64/libc.so (__start_thread+16)
2020-...art/runtime/runtime.cc:422]   (no managed stack frames)
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:422] 
2020-...art/runtime/runtime.cc:427] JNI ERROR (app bug): local reference table overflow (max=512)
2020-...art/runtime/runtime.cc:427] local reference table dump:
2020-...art/runtime/runtime.cc:427]   Last 10 entries (of 512):
2020-...art/runtime/runtime.cc:427]       511: 0x12ded740 java.lang.String ""
2020-...art/runtime/runtime.cc:427]       510: 0x12df90e8 com.emanuelef.remote_capture.ConnDescriptor[] (3 elements)
2020-...art/runtime/runtime.cc:427]       509: 0x12ca9d30 java.lang.Class<com.emanuelef.remote_capture.ConnDescriptor>
2020-...art/runtime/runtime.cc:427]       508: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/runtime.cc:427]       507: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/runtime.cc:427]       506: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/runtime.cc:427]       505: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/runtime.cc:427]       504: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/runtime.cc:427]       503: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/runtime.cc:427]       502: 0x12cb5800 java.lang.Class<com.emanuelef.remote_capture.CaptureService>
2020-...art/runtime/runtime.cc:427]   Summary:
2020-...art/runtime/runtime.cc:427]        53 of com.emanuelef.remote_capture.ConnDescriptor (53 unique instances)
2020-...art/runtime/runtime.cc:427]         1 of com.emanuelef.remote_capture.ConnDescriptor[]
2020-...art/runtime/runtime.cc:427]         5 of com.emanuelef.remote_capture.ConnDescriptor[] (2 elements) (5 unique instances)
2020-...art/runtime/runtime.cc:427]        14 of com.emanuelef.remote_capture.ConnDescriptor[] (3 elements) (14 unique instances)
2020-...art/runtime/runtime.cc:427]         1 of com.emanuelef.remote_capture.ConnDescriptor[] (4 elements)
2020-...art/runtime/runtime.cc:427]         1 of java.lang.Thread
2020-...art/runtime/runtime.cc:427]       274 of java.lang.String (274 unique instances)
2020-...art/runtime/runtime.cc:427]       163 of java.lang.Class (2 unique instances)
2020-...art/runtime/runtime.cc:427] 
2020-...art/runtime/runtime.cc:427] 
2020-...Fatal signal 6 (SIGABRT), code -6 in tid 4480 (CaptureService )

```

**Comment by emanuele-f** [comment]
Are you using the latest 1.2.2 release? Is there any particular condition to reproduce the crash?

**Comment by emanuele-f** [comment]
This should be fixed in 6d665bf9d8f12c4ecb04cfbcdcc2ae37b39282a7 . I'll release a new 1.2.3 version with the fix soon.

**Comment by emanuele-f** [comment]
@joshatcaper did you have a chance to test this? The 1.2.3 package is already available.

**Comment by joshatcaper** [comment]
Sorry @emanuele-f using this became a low priority at my company. Probably not going to have time to test this again until we all return to the office later next year. Sorry!

**Comment by emanuele-f** [comment]
Ok no problem, closing this for now
